src/ccl_neutrinos.c

```diff
@@ -95,114 +95,123 @@ INPUTS: a: scale factor, Neff: number of neutrino species, mnu: total mass in eV
-INPUTS: a: scale factor, Neff: number of neutrino species, OmNuh2: neutrino mass density Omeganu * h^2, TCMB: CMB temperature, accel: pointer to an accelerator which will evaluate the neutrino phasespace spline if defined, status: pointer to status integer.
-TASK: Compute Omeganu * h^2 as a function of time.
+INPUTS: OmNuh2: neutrino mass density today Omeganu * h^2, label: how you want to split up the masses, see ccl_neutrinos.h for options, TCMB: CMB temperature, accel: pointer to an accelerator which will evaluate the neutrino phasespace spline if defined, status: pointer to status integer.
+TASK: Given Omeganuh2 today, the method of splitting into masses, and the temperature of the CMB, output a pointer to the array of neutrino masses (may be length 1 if label asks for sum) 
 */
-double ccl_Omeganuh2_to_Mnu(double a, double Neff, double OmNuh2, double TCMB, gsl_interp_accel* accel, int* status){
-  // First check if Neff if 0
-  if (Neff==0) return 0.0;
-  
-  // Now handle the massless case
-  double Tnu, a4, prefix_massless,Omeganuh2_massless;
-
-  Tnu=TCMB*pow(4./11.,1./3.);
-  a4=a*a*a*a;  
-  prefix_massless = NU_CONST  * Tnu * Tnu * Tnu * Tnu; 
-  Omeganuh2_massless = Neff*prefix_massless*7./8./a4;
-  if ( OmNuh2 < Omeganuh2_massless) {
-    return 0.0;
-  }
-
-
-  int root_status, iter = 0, max_iter = 100;
-  const gsl_root_fsolver_type *T;
-  gsl_root_fsolver *s;
 
-  double m_root =-1., m_iter =0.;
-  double m_min = 1.e-12; //required for consistency with massless case in Omeganuh2
-  double m_max = 100.0; //root finding requires some upper bound
-  gsl_function F;
-  OmNuh2_params p;
-  p.a = a;
-  p.Neff =Neff;
-  p.OmNuh2_target = OmNuh2;
-  p.TCMB = TCMB;
-  p.accel = accel;
-  p.status = status;
-
-  F.function = &Omeganuh2_root;
-  F.params = &p;
-
-  T = gsl_root_fsolver_brent;
-  s = gsl_root_fsolver_alloc (T);
-  gsl_root_fsolver_set (s, &F, m_min, m_max);
-
-  do
-    {
-      iter++;
-      root_status = gsl_root_fsolver_iterate (s);
-      if (root_status){break;} //could not evaluate Omeganuh2
-      m_iter = gsl_root_fsolver_root (s);
-      m_min = gsl_root_fsolver_x_lower (s);
-      m_max = gsl_root_fsolver_x_upper (s);
-      root_status = gsl_root_test_interval (m_min, m_max,
-                                       0, 0.001); // double epsabs, double epsrel
-    }
-  while (root_status == GSL_CONTINUE && iter < max_iter && *status ==0);
-
-  if (root_status == GSL_SUCCESS){
-    m_root = m_iter;
-  }
-  else{*status = CCL_ERROR_NU_SOLVE;}
-  gsl_root_fsolver_free (s);
-
-  return m_root;
-}
+double* ccl_nu_masses(double OmNuh2, ccl_nu_masses_label label, double TCMB, gsl_interp_accel* accel, int* status){
+  
+  double sumnu;
+  
+  sumnu = 93.14 * OmNuh2;
+  
+  // Now split the sum up into three masses depending on the label given:
+  
+  if(label==ccl_nu_masses_normal_label){
+	  
+     double *mnu;
+	 mnu = malloc(3*sizeof(double));
+			
+	 if (sumnu>0.059){
+		 mnu[0] = (sumnu - 0.059) / 3.;
```

Comment:
Mention where these numbers are from. If they are measured physical parameters, they should probably be stored with the other physical constants, like G.

Reply:
I added them to the constants.h file. They're the experimentally measured mass splittings. I changed this a bit to make it clearer I think.